source/brailleDisplayDrivers/hims.py

```diff
@@ -278,11 +331,26 @@ def __init__(self, port="auto"):
 	def display(self, cells: List[int]):
 		# cells will already be padded up to numCells.
 		cellBytes = bytes(cells)
-		self._sendPacket(b"\xfc", b"\x01", cellBytes)
+		if self.isHID:
+			outputReport: bytes = b"".join([
+				intToByte(self.numCells),  # length
+				cellBytes
+			])
+
+			self._dev.setOutputReport(outputReport)
+		else:
+			self._sendPacket(b"\xfc", b"\x01", cellBytes)
 
 	def _sendCellCountRequest(self):
-		log.debug("Sending cell count request...")
-		self._sendPacket(b"\xfb", b"\x01", bytes(32)) # send 32 null bytes
+		if self.isHID:
+			try:
+				data: bytes = self._dev.getFeature(HR_CAPS)
+				self.numCells = data[9]
+			except WindowsError:
+				return  # Fail!

```

Comment:
please log something here, and make the comment more descriptive if necessary

```suggestion
				log.exception("Failed to fetch number of cells")
				return
```

Reply:
I think that log is enough. Thanks!